Emerald Insight.js

```diff
@@ -0,0 +1,240 @@
+
+	Copyright © 2019 Sebastian Karcher
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+// attr()/text() v2
+// eslint-disable-next-line
+function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
+
+
+function detectWeb(doc, url) {
+	if (url.includes('/content/doi/')) {
+		if (attr(doc, 'meta[name="dc.Type"]', 'content') == "book-part") {
+			return "bookSection";
+		}
+		else return "journalArticle";
+	}
+	else if (getSearchResults(doc, url, true)) {
+		return "multiple";
+	}
+	return false;
+}
+
+
+function getSearchResults(doc, url, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows;
+	if (url.includes("insight/search?")) {
+		// search results
+		rows = doc.querySelectorAll('h2>a.intent_link');
+	}
+	else {
+		// journal issue
+		rows = doc.querySelectorAll('.intent_issue_item h4>a');
+		if (!rows.length) {
+			// book
+			rows = doc.querySelectorAll('li.intent_book_chapter>a');
+		}
+	}
+	for (let i = 0; i < rows.length; i++) {
+		let href = rows[i].href;
+		let title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, url, false), function (items) {
+			if (!items) {
+				return;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var DOI = url.match(/\/(10\.[^#?/]+\/[^#?/]+)\//)[1];
+	var risURL = "/insight/content/doi/" + DOI + "/full/ris";
+	// Z.debug(risURL);
+	var pdfURL = attr(doc, 'a.intent_pdf_link', 'href');
+	// Z.debug("pdfURL: " + pdfURL);
+	ZU.doGet(risURL, function (text) {
+		// they number authors in their RIS...
+		text = text.replace(/A\d+\s+-/g, "AU  -");
+		// add a comma after the last name
+		text = text.replace(/AU\s\s-\s(\w+)/g, "AU  - $1,");
```

Comment:
I think we should test here first that no comma is present at the moment (because they might fix that at some point). Moreover, can you extend the comment for this a little more?

It seems to me that they are always treating the last word of the author string as the last name and everything else as the first name(s), which they then put into RIS data as "lastname firstname(s)" without a comma. The splitting does not need to be correct, e.g. https://www.emerald.com/insight/content/doi/10.1108/S1572-832320170000026008/full/html , but it is the only thing we can get from their metadata.

Reply:
Done and done.